File: src/main/java/com/ptaf/ai/config/LocalDotEnv.java
package com.ptaf.ai.config;

import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

/**
 * Loads {@code .env} from the process working directory (project root when using Maven exec).
 * Does not override real environment variables — use {@link AiAssistantProperties} for merge order.
 */
final class LocalDotEnv {

    private static final Map<String, String> ENTRIES = load();

    private LocalDotEnv() {
    }

    static String get(String key) {
        if (key == null) {
            return null;
        }
        return ENTRIES.get(key);
    }

    private static Map<String, String> load() {
        Path p = Path.of(System.getProperty("user.dir", ".")).resolve(".env").normalize();
        if (!Files.isRegularFile(p)) {
            return Collections.emptyMap();
        }
        Map<String, String> map = new HashMap<>();
        try {
            for (String line : Files.readAllLines(p, StandardCharsets.UTF_8)) {
                String t = line.trim();
                if (t.isEmpty() || t.startsWith("#")) {
                    continue;
                }
                int eq = t.indexOf('=');
                if (eq <= 0) {
                    continue;
                }
                String k = t.substring(0, eq).trim();
                String v = t.substring(eq + 1).trim();
                if (v.startsWith("\"") && v.endsWith("\"") && v.length() >= 2) {
                    v = v.substring(1, v.length() - 1);
                } else if (v.startsWith("'") && v.endsWith("'") && v.length() >= 2) {
                    v = v.substring(1, v.length() - 1);
                }
                if (!k.isEmpty()) {
                    map.put(k, v);
                }
            }
        } catch (IOException e) {
            return Collections.emptyMap();
        }
        return Collections.unmodifiableMap(map);
    }
}

File: ptaf/ai/config/local_dot_env.py
"""Loads .env from the process working directory."""

from __future__ import annotations

import os
from pathlib import Path


def _load() -> dict[str, str]:
    env_path = Path(os.getcwd()).resolve() / ".env"
    if not env_path.is_file():
        return {}
    entries: dict[str, str] = {}
    for line in env_path.read_text(encoding="utf-8").splitlines():
        text = line.strip()
        if not text or text.startswith("#"):
            continue
        if "=" not in text:
            continue
        key, _, value = text.partition("=")
        key = key.strip()
        value = value.strip()
        if len(value) >= 2 and (
            (value.startswith('"') and value.endswith('"'))
            or (value.startswith("'") and value.endswith("'"))
        ):
            value = value[1:-1]
        if key:
            entries[key] = value
    return entries


_ENTRIES = _load()


def get(key: str | None) -> str | None:
    if key is None:
        return None
    return _ENTRIES.get(key)
